### ISSUE TITLE
Restore tab position/location when reopening closed tab

### ISSUE THREAD

**Issue by vertigo220** [notepad-plus-plus/notepad-plus-plus#13043]
### Description of the Issue

If a bunch of tabs are open and some are accidentally closed, reopening them just puts them at the end of the tabs instead of back where they were, making it more difficult to keep track.

### Steps to Reproduce the Issue

1. Open several tabs (or more)
2. Close one or more tabs
3. Reopen one or more tabs

### Expected Behavior

For the tabs to be reopened in their original position in the tab bar, similar to a web browser

### Actual Behavior

Tabs are reopened and placed at the end of the tab bar

### Debug Information

Notepad++ v8.4.8   (64-bit)
Build time : Dec 24 2022 - 19:40:27
Path : C:\Program Files\Notepad++\notepad++.exe
Command Line : "D:\Programs\Multimedia--Photo Editing\Video\Players\PotPlayer\!Backup Config ~RUN AS ADMIN~.bat" "D:\Programs\Multimedia--Photo Editing\Video\Players\PotPlayer\!Post-Install Setup (Config Restore) ~RUN AS ADMIN~.bat"
Admin mode : OFF
Local Conf mode : OFF
Cloud Config : OFF
OS Name : Windows 10 Pro N (64-bit) 
OS Version : 22H2
OS Build : 19045.2486
Current ANSI codepage : 1252
Plugins : 
    mimeTools (2.9)
    NavigateTo (1.12.4.1)
    NppConverter (4.5)
    NppExport (0.4)
    NppMenuSearch (0.9.3)
    NppSaveAsAdmin (1.0.211)
    NppSnippets (1.6)
    RegexTrainer (1)
    RunMe (1.4)
    SurroundSelection (1.4.1)
    _CustomizeToolbar (4.2)

**Comment by alankilborn** [comment]
Nah, Notepad++ 's design is that it doesn't remember anything about closed tabs; too much burden to do so.

**Comment by vertigo220** [comment]
It remembers their name and location, and it maintains info on all the open tabs and their positions, so all it would have to do would be to flag a tab in that list when it's closed as closed, so it won't reopen it e.g. when Notepad++is closed and reopened, but if that file were reopened via the "Restore Recent Closed File" command it could see where it was from that. And these flagged tabs could drop off as new ones are flagged, so it only remembers the last x amount, with x being the max number of entries in Preferences > Recent Files History. That should should add a negligible burden.

**Comment by alankilborn** [comment]
And it only makes sense, really, if none of the other open tabs change their position, and brand new tabs aren't opened in the interim; otherwise, how can the ones you want be restored to where they were...  Overall:  Blech!

**Comment by camileevzw** [comment]
Agreed, @alankilborn . I was just going to point out that if tab#3 is supposed to be repositioned, how would that possibly work if tabs 1 and 2 were closed an not desired to be reopened? That kind of creature feature would be trouble waiting for a chance to happen. In addition, that extra bookkeeping would probably bog NPP down and be even more cumbersome to open  up. Imaging all the tabs ever being closed being documented and having to open that file along with NPP opening in general.  Basically, it's a niche need, when all the user has to do, and what I already do, is reopen the document, and reposition the tab to where I want it, in the order I had it.

**Comment by alankilborn** [comment]
> Imagine all the tabs ever being closed being documented

Well, OP did limit his request:

>...it only remembers the last x amount,...
